### ISSUE TITLE
Doc: Database backup copy order of chk files.

### ISSUE THREAD

**Issue by jezovuk** [EventStore/EventStore#2209]
Hi,

I have a question regarding backup order of *.chk files themselves.

If I understand various bits of info from documentation, Github issues and Google Groups, content of chaser.chk is pointer to position (in chunk files "continuum") reached by chaser process, while writer.chk is pointer to position reached by writer process. Chaser process should always be behind the writer (or at the same point), never ahead of it.
Does it make sense to copy chaser.chk before writer.chk when performing backup?
The docs only say that all *.chk files should be copied before all other files (index and data chunks), but does not specify ordering between various *.chk files. From what I understand, if writer.chk is copied before chaser.chk, it might be possible that copied chaser.chk contains pointer that is after (larger than) value in copied writer.chk, which might be a problem (not sure if EventStore can recover from this during startup).
Also, another unclear point: what is the purpose of epoch.chk and its relation to writer.chk and chaser.chk? Does it matter if it is copied before or after writer/chaser chk files? (Info on epoch.chk is particularly hard to find online).
As for truncate.chk, it will be overwritten by chaser.chk during restore, so it really doesn't matter when it gets copied (or if at all).

To sum it up in one question: what should the order of copying chk files during backup be so that backup consistency is ensured? (Explicit ordered list would be really helpful and reassuring).
It would be ideal if backup page in docs elaborated on this a bit.

Note that I'm mostly interested in single-node setup.

**Comment by timothycoleman** [comment]
Hi @jezovuk, the backup documentation has been elaborated on since you asked your question

https://developers.eventstore.com/server/v21.10/operations/database-backup.html#differential-backup-restore

an explicit order that works is chaser, epoch, writer, proposal

the purpose of the epoch checkpoint is to point to the last epoch record in the log. the epochs themselves point to the previous epoch so the rest of the epochs can be found quickly after this. the epochs are used when a node joins its cluster to ensure it has the right data. the database can startup without this checkpoint but it will scan the log backwards until it finds an epoch record (which could be a long way if the node was running stably for a long time)
